File: Calculator/android-native/app/src/main/java/com/qaantum/calculatorhub/calculators/Converters.kt
package com.qaantum.calculatorhub.calculators

// Unit Converter - Multiple types
class UnitConverter {
    // Length
    fun metersToFeet(m: Double) = m * 3.28084
    fun feetToMeters(ft: Double) = ft / 3.28084
    fun kmToMiles(km: Double) = km * 0.621371
    fun milesToKm(mi: Double) = mi / 0.621371
    fun cmToInches(cm: Double) = cm / 2.54
    fun inchesToCm(inches: Double) = inches * 2.54

    // Weight
    fun kgToLbs(kg: Double) = kg * 2.20462
    fun lbsToKg(lbs: Double) = lbs / 2.20462
    fun gramsToOz(g: Double) = g * 0.035274
    fun ozToGrams(oz: Double) = oz / 0.035274

    // Temperature
    fun celsiusToFahrenheit(c: Double) = c * 9 / 5 + 32
    fun fahrenheitToCelsius(f: Double) = (f - 32) * 5 / 9
    fun celsiusToKelvin(c: Double) = c + 273.15
    fun kelvinToCelsius(k: Double) = k - 273.15

    // Volume
    fun litersToGallons(l: Double) = l * 0.264172
    fun gallonsToLiters(gal: Double) = gal / 0.264172
    fun mlToFlOz(ml: Double) = ml * 0.033814
    fun flOzToMl(oz: Double) = oz / 0.033814

    // Area
    fun sqMetersToSqFeet(m2: Double) = m2 * 10.7639
    fun sqFeetToSqMeters(ft2: Double) = ft2 / 10.7639
    fun acresToHectares(acres: Double) = acres * 0.404686
    fun hectaresToAcres(ha: Double) = ha / 0.404686

    // Speed
    fun kphToMph(kph: Double) = kph * 0.621371
    fun mphToKph(mph: Double) = mph / 0.621371
    fun msToKph(ms: Double) = ms * 3.6
    fun kphToMs(kph: Double) = kph / 3.6

    // Data Storage
    fun bytesToKB(bytes: Double) = bytes / 1024
    fun kbToMB(kb: Double) = kb / 1024
    fun mbToGB(mb: Double) = mb / 1024
    fun gbToTB(gb: Double) = gb / 1024
}

// Pressure Converter
class PressureConverter {
    fun psiToBar(psi: Double) = psi * 0.0689476
    fun barToPsi(bar: Double) = bar / 0.0689476
    fun atmToPa(atm: Double) = atm * 101325
    fun paToAtm(pa: Double) = pa / 101325
    fun mmHgToAtm(mmHg: Double) = mmHg / 760
}

// Power Converter
class PowerConverter {
    fun wattsToHP(w: Double) = w / 745.7
    fun hpToWatts(hp: Double) = hp * 745.7
    fun wattsToKw(w: Double) = w / 1000
    fun kwToBTU(kw: Double) = kw * 3412.14
}

// Angle Converter
class AngleConverter {
    fun degreesToRadians(deg: Double) = deg * Math.PI / 180
    fun radiansToDegrees(rad: Double) = rad * 180 / Math.PI
    fun degreesToGradians(deg: Double) = deg * 10 / 9
    fun gradiansToDegrees(grad: Double) = grad * 9 / 10
}

// Torque Converter
class TorqueConverter {
    fun nmToLbFt(nm: Double) = nm * 0.737562
    fun lbFtToNm(lbft: Double) = lbft / 0.737562
    fun nmToKgCm(nm: Double) = nm * 10.1972
}

// Fuel Consumption Converter
class FuelConverter {
    fun mpgToLper100km(mpg: Double) = 235.215 / mpg
    fun lper100kmToMpg(l100km: Double) = 235.215 / l100km
    fun mpgUKToMpgUS(mpgUK: Double) = mpgUK * 0.832674
}

// Shoe Size Converter
class ShoeSizeConverter {
    fun usToEU(usSize: Double, isMale: Boolean): Double = if (isMale) usSize + 33 else usSize + 31
    fun euToUS(euSize: Double, isMale: Boolean): Double = if (isMale) euSize - 33 else euSize - 31
    fun usToUK(usSize: Double, isMale: Boolean): Double = if (isMale) usSize - 0.5 else usSize - 2
}

// Cooking Converter
class CookingConverter {
    fun cupsToMl(cups: Double) = cups * 236.588
    fun mlToCups(ml: Double) = ml / 236.588
    fun tbspToMl(tbsp: Double) = tbsp * 14.787
    fun tspToMl(tsp: Double) = tsp * 4.929
    fun ozToGrams(oz: Double) = oz * 28.3495
}

File: Calculator/ios-native/CalculatorHub/Calculators/Converters.swift
import Foundation

// Unit Converter
class UnitConverter {
    // Length
    func metersToFeet(_ m: Double) -> Double { m * 3.28084 }
    func feetToMeters(_ ft: Double) -> Double { ft / 3.28084 }
    func kmToMiles(_ km: Double) -> Double { km * 0.621371 }
    func milesToKm(_ mi: Double) -> Double { mi / 0.621371 }
    func cmToInches(_ cm: Double) -> Double { cm / 2.54 }
    func inchesToCm(_ inches: Double) -> Double { inches * 2.54 }

    // Weight
    func kgToLbs(_ kg: Double) -> Double { kg * 2.20462 }
    func lbsToKg(_ lbs: Double) -> Double { lbs / 2.20462 }

    // Temperature
    func celsiusToFahrenheit(_ c: Double) -> Double { c * 9 / 5 + 32 }
    func fahrenheitToCelsius(_ f: Double) -> Double { (f - 32) * 5 / 9 }
    func celsiusToKelvin(_ c: Double) -> Double { c + 273.15 }

    // Volume
    func litersToGallons(_ l: Double) -> Double { l * 0.264172 }
    func gallonsToLiters(_ gal: Double) -> Double { gal / 0.264172 }

    // Area
    func sqMetersToSqFeet(_ m2: Double) -> Double { m2 * 10.7639 }
    func acresToHectares(_ acres: Double) -> Double { acres * 0.404686 }

    // Speed
    func kphToMph(_ kph: Double) -> Double { kph * 0.621371 }
    func mphToKph(_ mph: Double) -> Double { mph / 0.621371 }

    // Data
    func bytesToKB(_ bytes: Double) -> Double { bytes / 1024 }
    func mbToGB(_ mb: Double) -> Double { mb / 1024 }
}

// Pressure Converter
class PressureConverter {
    func psiToBar(_ psi: Double) -> Double { psi * 0.0689476 }
    func barToPsi(_ bar: Double) -> Double { bar / 0.0689476 }
    func atmToPa(_ atm: Double) -> Double { atm * 101325 }
}

// Power Converter
class PowerConverter {
    func wattsToHP(_ w: Double) -> Double { w / 745.7 }
    func hpToWatts(_ hp: Double) -> Double { hp * 745.7 }
    func kwToBTU(_ kw: Double) -> Double { kw * 3412.14 }
}

// Angle Converter
class AngleConverter {
    func degreesToRadians(_ deg: Double) -> Double { deg * .pi / 180 }
    func radiansToDegrees(_ rad: Double) -> Double { rad * 180 / .pi }
}

// Torque Converter
class TorqueConverter {
    func nmToLbFt(_ nm: Double) -> Double { nm * 0.737562 }
    func lbFtToNm(_ lbft: Double) -> Double { lbft / 0.737562 }
}

// Fuel Consumption Converter
class FuelConverter {
    func mpgToLper100km(_ mpg: Double) -> Double { 235.215 / mpg }
    func lper100kmToMpg(_ l100km: Double) -> Double { 235.215 / l100km }
}

// Shoe Size Converter
class ShoeSizeConverter {
    func usToEU(_ usSize: Double, isMale: Bool) -> Double { isMale ? usSize + 33 : usSize + 31 }
    func euToUS(_ euSize: Double, isMale: Bool) -> Double { isMale ? euSize - 33 : euSize - 31 }
}

// Cooking Converter
class CookingConverter {
    func cupsToMl(_ cups: Double) -> Double { cups * 236.588 }
    func mlToCups(_ ml: Double) -> Double { ml / 236.588 }
    func tbspToMl(_ tbsp: Double) -> Double { tbsp * 14.787 }
    func tspToMl(_ tsp: Double) -> Double { tsp * 4.929 }
}
